### DISCUSSION TITLE
What's the easiest way to find the number of unique :user/:repo repositories that use a new language?

### DISCUSSION THREAD

**Discussion by hirehamir** [github-linguist/linguist#5897] [Q&A] [answered]
Source files for the new language in question end with a `.rsh` extension.

The first line of all programs in this language as of Saturday, May 14, 2022, should contain `reach 0.1`.

I tried using the GitHub API to process `response`s with
```ts
    const response = await octokit.request(
      `GET /search/code?q=reach+0.1+extension%3Arsh&per_page=100&page=${
        pageNumber
      }`
    );
```
but I run into `RequestError [HttpError]: Only the first 1000 search results are available` and
```
    data: {
      message: 'Only the first 1000 search results are available',
      documentation_url: 'https://docs.github.com/v3/search/'
    }
```

[There are 2,057 code results when I enter this search query through the browser](https://github.com/search?q=reach+0.1+extension%3Arsh&type=Code&ref=advsearch&l=&l=), as of Saturday, May 14, 2022.

The issue is that I don't know how to download all of these search results as a `json` so I can do some processing on it to find the number of unique users. Trying to do this via the GitHub search API caps me at the first 1,000 search results: I want to analyze all 2,000+ results.

I'm trying to find out whether this language in in use by at least 200 unique users, not counting forks, because [the `Adding an extension to a language` section of `CONTRIBUTING.md`](https://github.com/github/linguist/blob/master/CONTRIBUTING.md#adding-an-extension-to-a-language) says

> In most cases we prefer that each new file extension be in use in at least 200 unique :user/:repo repositories before supporting them in Linguist.

How are other people checking this?

**Comment by lildude** [comment, marked as answer]
We've historically used [Harvester](https://github.com/Alhadis/Harvester) or a script I've written which does the same thing but using the API, however we're temporarily not using this as GitHub's Search is struggling. See https://github.com/github/linguist/issues/5756 for more details on the temporary requirements.
